discord/http.py

```diff
@@ -942,6 +942,18 @@ def get_reaction_users(
         if after:
             params['after'] = after
         return self.request(r, params=params)
+    
+    def get_guild_shop(
+        self,
+        guild_id: Snowflake
+    ) -> Response[Optional[guild.GuildShop]]:
```

Comment:
```suggestion
    ) -> Response[guild.GuildShop]:
```
I find it very unlikely that the json payload from a GET request will be a literal `null` in the event that the guild shop does not exist.

Much more likely you're gonna get some kind of error response which should be raised as an exception.

Reply:
Yep, I also changed that.